inc/connector/namespace.php

```diff
@@ -11,6 +14,351 @@
+ *                   events were registered with the hook and arguments combination), false if
+ *                   unscheduling one or more events fail.
+*/
+function pre_clear_scheduled_hook( $pre, $hook, $args ) {
+
+	// First check if the job exists already.
+	$jobs = Job::get_jobs_by_query(
+		[
+			'hook' => $hook,
+			'args' => $args,
+			'limit' => 100,
+		]
+	);
+
+	if ( empty( $jobs ) ) {
+		// No jobs to unschedule.
+		return 0;
+	}
+
+	$ids = wp_list_pluck( $jobs, 'id' );
+
+	global $wpdb;
+
+	// Clear all scheduled events for this site
+	$table = Job::get_table();
+
+	$sql = "DELETE FROM `{$table}` WHERE site = %d";
+	$sql_params[] = get_current_blog_id();
+
+	$sql .= ' AND id IN(' . implode( ',', array_fill( 0, count( $ids ), '%d' ) ) . ')';
+	$sql_params = array_merge( $sql_params, $ids );
+
+	$query = $wpdb->prepare( $sql, $sql_params );
+	$results = $wpdb->query( $query );
+
+	// Flush the caches.
+	Job::flush_query_cache();
+	array_walk( $ids, function( $id ) {
+		wp_cache_delete( "job::{$id}", 'cavalcade-jobs' );
+	} );
+
+	return $results;
+}
+
+/**
+ * Unschedules all events attached to the hook.
+ *
+ * Can be useful for plugins when deactivating to clean up the cron queue.
+ *
+ * Warning: This function may return Boolean FALSE, but may also return a non-Boolean
+ * value which evaluates to FALSE. For information about casting to booleans see the
+ * {@link https://php.net/manual/en/language.types.boolean.php PHP documentation}. Use
+ * the `===` operator for testing the return value of this function.
+ *
+ * @param null|array $pre  Value to return instead. Default null to continue unscheduling the hook.
+ * @param string     $hook Action hook, the execution of which will be unscheduled.
+ * @return bool|int On success an integer indicating number of events unscheduled (0 indicates no
+ *                  events were registered on the hook), false if unscheduling fails.
+ */
+function pre_unschedule_hook( $pre, $hook ) {
+	return pre_clear_scheduled_hook( $pre, $hook, null );
+}
+
+/**
+ * Retrieve a scheduled event.
+ *
+ * Retrieve the full event object for a given event, if no timestamp is specified the next
+ * scheduled event is returned.
+ *
+ * @param null|bool $pre       Value to return instead. Default null to continue retrieving the event.
+ * @param string    $hook      Action hook of the event.
+ * @param array     $args      Array containing each separate argument to pass to the hook's callback function.
+ *                             Although not passed to a callback, these arguments are used to uniquely identify the
+ *                             event.
+ * @param int|null  $timestamp Unix timestamp (UTC) of the event. Null to retrieve next scheduled event.
+ * @return bool|object The event object. False if the event does not exist.
+ */
+function pre_get_scheduled_event( $pre, $hook, $args, $timestamp ) {
+	return $pre;
```

Comment:
@peterwilsoncc only just noticed this! Do we actually need the hook at all?

Reply:
We do, that's a screw up by yours truly.